### PR TITLE
Many improvements, addressing several open issues

### PR DESCRIPTION
**Pull request by superzanti**
- Added visual mode functionality
- Added before paste and after paste functionality ("+p and "+P)
- Fixed pathing issues and added an option for correct pathing
- Added a default filetype so you can paste anywhere
- Fixed buffer of different filetype issue
- Added ability to use a Lua function for the filetype (for things like date-time formatting)

Issues fixed by this code:

1. [how to set the img name to date-time format without set name everytime](https://github.com/TobinPalmer/pastify.nvim/issues/4)
2. [https://github.com/TobinPalmer/pastify.nvim/issues/7](https://github.com/TobinPalmer/pastify.nvim/issues/7)
3. [https://github.com/TobinPalmer/pastify.nvim/issues/8](https://github.com/TobinPalmer/pastify.nvim/issues/8)

### CODE REVIEW

**Comment by TobinPalmer** [comment]
Wow I cannot express my gratitude for this amazing PR. I’ve been very busy with school and finals and this is such a big pr solving most of the open issues. You are the reason I love the open source community ❤️

**Comment by superzanti** [comment]
Thank you for the gratitude! I was worried I did too much in one commit and it might be too hard to follow.
Thanks for the merge :)
